Minor fixes

soreau opened the pull request.

soreau committed:
install.sh: Don't use sudo if sudo isn't available

This might be the case on systems like debian.

soreau committed:
install.sh: Set CAP_SYS_ADMIN if neither systemd nor elogind is found

soreau committed:
install.sh: Install start_wayfire.sh as /bin/startwayfire

ammen99 commented on install.sh:
```diff
@@ -103,24 +103,29 @@ check_download() {
 
     # Checkout the correct stream
     cd $1
-    git checkout origin/${STREAM}
```
Why this? I want to check out the upstream branch, not local one.

soreau replied:
Doing this doesn't checkout the upstream branch, it leaves the repo in a detached head state.

ammen99 replied:
It checkouts the upstream branch, which is exactly what a detached head state is.

ammen99 replied:
Well not exactly, but `origin/${STREAM}` is what I meant and I think it should stay this way.

soreau replied:
Is there a way you can skip this patch or would I have to take the patch out of the series?

ammen99 replied:
You can just rebase and drop the patch.

ammen99 commented on install.sh:
```diff
@@ -103,24 +103,29 @@ check_download() {
 
 cd $BUILDROOT/wayfire
-
+[ -d build ] && rm -rf build
```
Why this? Didn't you want to have possibility for kind of incremental build, if some stage of the script fails? If the directories are to be deleted, then the user is supposed to use --clean.

soreau replied:
It still reruns meson and ninja when rerunning the script. If it has failed previously, the next time it is run, meson will error that the directory is already configured. I believe there is some caching even when the build directory is deleted, so subsequent runs are quicker.

ammen99 replied:
The caching comes from ccache, if anything. Maybe we should always do a clean build then, and drop the option.

ammen99 commented on install.sh:
```diff
@@ -66,7 +66,7 @@ fi
 echo "Installation prefix: $PREFIX"
 
-BUILDROOT=$(pwd)
+BUILDROOT="${0%/*}"
```
What does this do?

soreau replied:
If for instance you clone wf-install and then run `./wf-install/install.sh` then $(pwd) will be `.` and when it goes to find files from the build root, specifically the start script .in, build root will not be the repo directory, so it doesn't find the file. The 0 in "${0%/*}" would be `./wf-install/install.sh` and it eliminates everything after and including the last forward slash. So it would evaluate to `./wf-install`. This works regardless which directory it is run from.

ammen99 replied:
-1, the user should go where they want to build Wayfire.

soreau committed:
install.sh: Use path where the script lives as the build root

Using /home/scott/src/wayland/wayfire/wf-install assumes the script is being run from the repo directory.
Instead we should get the path by changing into dirname $0 directory
first and then using pwd -P.

ammen99 commented:
Thanks for these patches!
